justfile

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env just --justfile
+# ^ A shebang isn't required, but allows a justfile to be executed
+#   like a script, with `./justfile test`, for example.
+
+ROOT := justfile_directory()
+DOCS := join(ROOT, "docs")
+MAN_DIR := join(DOCS, "man-page")
+MAN_MD := join(MAN_DIR, "py.1.md")
+MAN_FILE := join(MAN_DIR, "py.1")
+CARGO_TOML := join(ROOT, "Cargo.toml")
+DOT_DIR := join(DOCS, "control-flow")
+DOT_FILE := join(DOT_DIR, "control_flow.dot")
+DOT_STEM := file_stem(DOT_FILE)
+DOT_SVG := join(DOT_DIR, DOT_STEM) + ".svg"
+DOT_PNG := join(DOT_DIR, DOT_STEM) + ".png"
+
+# TODO: Next just release will make `join` accept variadic parameters
+# would clean up the variables quite a bit
+
+# Set default recipes
+_default: lint test man dot
+
+# Run the unit tests
+test:
+    cargo --quiet test
+
+# Run linting on source files
+lint:
+    cargo fmt --quiet --all -- --check
+    cargo clippy --quiet --all-targets --all-features -- -D warnings
+
+# Install from source
+install:
+    cargo install --quiet --path .
+
+# Convert the markdown man to man file format (hidden)
+_man-md:
+    pandoc {{ MAN_MD }} --standalone -t man -o {{ MAN_FILE }}
+
+# Build the man page
+man: _man-md
+    #!/usr/bin/env python3
+
+    import datetime
+    import pathlib
+    import re
+
+    with open("{{ CARGO_TOML }}", "r", encoding="utf-8") as file:
+        cargo_lines = file.readlines()
+
+    for line in cargo_lines:
+        version_match = re.match(r'version\s*=\s*"(?P<version>[\d.]+)"', line)
+        if version_match:
```

Comment:
```suggestion
        if version_match := VERSION_REGEX.match(line):
```

Reply:
Do you think it's worth explaining somewhere that a contributor would need python >=3.8 (I think!) in order to run this recipe or is it mainly just for CI where you control the version entirely?

Reply:
The failure for folks not running on Python 3.8 should be obvious. And if even Debian ships with 3.8 then it's reasonable to accept folks will have it installed.

Reply:
Cool sounds good to me 🙂